### PR TITLE
azure: update docs for Azure Container Service Kubernetes support

### PR DESCRIPTION
**Pull request by colemickens**
Update Azure docs:
* remove legacy Weave, CoreOS solution. (Had agreement from @aronchick and @errordeveloper that it's okay to remove this)
* point users to ACS and ACS-Engine as the encouraged bring-up method instead of Kubernetes-Anywhere

~~NOTE: Do not merge yet, this note will be removed when ready to merge. I'm mostly using this for the auto-generated per-PR rendered docs.~~

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes.github.io/1683)

### CODE REVIEW

**Comment by colemickens** [comment]
For now this will just link to the generic ACS docs. When the k8s walkthrough is published, we can link to that.

**Comment by colemickens** [comment]
I did this in a manner which mirrors GKE. GKE and GCE each have their own entires, with the former being a hotlink to the relevant GKE docs. For now the ACS link will just link to the generic ACS docs. When the k8s walkthrough is published, I will updated to link directly to the K8s on ACS walkthrough documentation.

**Comment by colemickens** [comment]
ping @brendandburns for a review/merge.

File: docs/getting-started-guides/coreos/azure/index.md [hunk 1]
```diff
@@ -1,246 +0,0 @@
...
-This script will provision a cluster suitable for production use, where there is a ring of 3 dedicated etcd nodes: 1 kubernetes master and 2 kubernetes nodes. The `kube-00` VM will be the master, your work loads are only to be deployed on the nodes, `kube-01` and `kube-02`. Initially, all VMs are single-core, to ensure a user of the free tier can reproduce it without paying extra. I will show how to add more bigger VMs later.
-If you need to pass Azure specific options for the creation script you can do this via additional environment variables e.g.
-
-```shell
-AZ_SUBSCRIPTION=<id> AZ_LOCATION="East US" ./create-kubernetes-cluster.js
-# or
-AZ_VM_COREOS_CHANNEL=beta ./create-kubernetes-cluster.js
-```
-
-![VMs in Azure](/images/docs/initial_cluster.png)
```
Comment (errordeveloper) [inline docs/getting-started-guides/coreos/azure/index.md]: Please remove the file from the repo also.

Comment (colemickens) [inline docs/getting-started-guides/coreos/azure/index.md]: It is already removed. :) The entire `azure` dir under `getting-started-guides/coreos` is removed in this PR.

File: docs/getting-started-guides/coreos/azure/index.md [hunk 2]
```diff
@@ -1,246 +0,0 @@
----
```
Comment (errordeveloper) [inline docs/getting-started-guides/coreos/azure/index.md]: We may want to keep the URL, WDYT? We can point people at https://github.com/weaveworks-guides/weave-kubernetes-coreos-azure... but actually, do people still use bookmarks? If anyone release used it, they'd know where to find the repo. Personally, I don't care too much, but I've briefly touch on removal of outdated turn-up docs with some folks at the dev summit, and heard that redirects are desired. (See also https://github.com/kubernetes/kubernetes.github.io/issues/1659).

Comment (colemickens) [inline docs/getting-started-guides/coreos/azure/index.md]: Hm, I don't have a strong opinion, but I would hesitate to point people to the weaveworks-guide unless it's something you plan to support (and I'm guessing it doesn't have the cloudprovider wired up). The supportability aspect is part of the reason I want to remove it; even as of last month people were asking questions about it that I couldn't necessarily answer.

I'd be less hesitant about preserving the URL and putting a note here that encourages people to seek out an alternative Azure method. (Though none of the "current" bring-up methods support CoreOS or Weave...)

**Comment by colemickens** [comment]
cc: @devin-donnelly Can you advise on how I can get this merged?

File: docs/getting-started-guides/coreos/azure/index.md (hunk 1 above)
Comment (devin-donnelly) [inline docs/getting-started-guides/coreos/azure/index.md]: Once the file in the root directory gets removed (per @errordeveloper ), I can merge this for you.

Comment (colemickens) [inline docs/getting-started-guides/coreos/azure/index.md]: Oh, I finally understand. Fixed, pushed.

**Comment by colemickens** [comment]
Alright, should be good to go.

**Comment by colemickens** [comment]
(ping for lgtm @devin-donnelly. thanks!)

**Comment by colemickens** [comment]
(ping for lgtm @devin-donnelly. thanks!)

**Comment by colemickens** [comment]
@davidopp Any chance you can merge this one?

**Comment by davidopp** [comment]
My understanding is that only someone from the tech writing team is supposed to merge docs PRs. 

ping @devin-donnelly

**Comment by colemickens** [comment]
ping @devin-donnelly It'd be great if we can get this merged soon. People are still picking up kube-anywhere for Azure despite our preference that they use ACS or ACS-Engine. I suspect that's due to the official kubernetes docs not being updated.

Thanks!

**Comment by colemickens** [comment]
ping @devin-donnelly

**Comment by brendandburns** [comment]
@devin-donnelly friendly ping on this.  Can you get this merged?

Thanks!
--brendan

**Comment by brendandburns** [comment]
@devin-donnelly this is rebased and ready to go...

**Pull request merged**